### PR TITLE
[WIP] [pr2eus_tutorials] Select grasp position via 2D image

### PR DESCRIPTION
**Pull request by 708yamaguchi**
Closes https://github.com/jsk-ros-pkg/jsk_pr2eus/issues/436

In this pull request, I did
- reuse `pr2_tabletop.launch` in `pr2_tabletop_grasp.launch` to simplify codes
- add gazebo mode in `pr2_tabletop_grasp.launch`
- enable to select grasp position via 2D image using `image_view2` and `PointcloudScreenpoint`
- update usage of these new features in README

[WIP] This pull request needs to be verified on real PR2 robot.

Here is sample implementation of this program:
https://drive.google.com/file/d/1JbG7HTPiQ2x-l756z9GTLHitZslFYvxe/view?usp=sharing

### CODE REVIEW

**Comment by knorth55** [comment]
first of all, have you tried xurrent implementation of gazebo version?

File: pr2eus_tutorials/README.md [hunk 1]
```diff
@@ -109,6 +109,15 @@ It means the robot now recognizes each objects as individual objects from camera
...
 
+##### Run all programs by one launch file
+```bash
+# Using a real PR2 robot
+ssh <robot address>
+roslaunch pr2eus_tutorials pr2_tabletop_grasp.launch
+# Using a simulator on local machine
+roslaunch pr2eus_tutorials pr2_tabletop_grasp.launch gazebo:=true
+```
+
```
Comment (knorth55) [inline pr2eus_tutorials/README.md:120]: `pr2_tabletop_grasp.launch` is deprecated.
in current implementation, we use `pr2_tabletop.launch` for the sample launch.
please merge use the launch, implement to use both rviz and image_view2 simaltaneously, and do not use `pr2_tabletop_grasp.launch`.

Comment (knorth55) [reply pr2eus_tutorials/README.md:120]: i misunderstood.
please run both rviz and image_view2 simaltaneously and do not use argument `grasp_by_2d_point`.

**Comment by 708yamaguchi** [comment]
Thank you for your advice.

I updated this pull request based on your advice (https://github.com/jsk-ros-pkg/jsk_pr2eus/issues/436 and https://github.com/jsk-ros-pkg/jsk_pr2eus/pull/437 ).

Also, I confirmed that both of `Rviz` click demo and `image_view2` click demo can be executed on gazebo simulation.

**Comment by k-okada** [comment]
great! can you update visualizatoin
1) when use click the image_view, add circle at that position (for a few
seconds, if possible)
2) visualize ray line and target 3D point,
ex)
https://jsk-recognition.readthedocs.io/en/latest/jsk_pcl_ros/nodes/pointcloud_screenpoint.html


--
◉ Kei Okada



2020年5月13日(水) 7:43 Naoya Yamaguchi <notifications@github.com>:

**Comment by 708yamaguchi** [comment]
> 1) when use click the image_view, add circle at that position (for a few
> seconds, if possible)
> 2) visualize ray line and target 3D point,

I pushed additional commits.

When we click `image_view2` window,
- green circle is shown for 5 seconds around the clicked pixel
- 3d ray to the target object is shown in Rviz for 5 seconds

For the second feature, I try to reuse https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_pcl_ros/euslisp/pointcloud_screenpoint.l
However, this script assumes the service version of `PointcloudScreenpoint` (I use the topic version in this pull request)
Also, some functions are automatically executed when we load this script. 
Therefore, I gave up reusing this script and left a comment.
```
;; Mainly copied from https://github.com/jsk-ros-pkg/jsk_recognition/blob/master/jsk_pcl_ros/euslisp/pointcloud_screenpoint.l
(defun show-marker (frame p1)
```

Sample
Screenshot 2020-05-13 22:10:30

**Comment by k-okada** [comment]
great work!

**Pull request merged**